### PR TITLE
Redux devtools integration

### PR DESCRIPTION
**Pull request by heldr**
Partialy based on https://github.com/QubitProducts/tiny-atom/blob/master/devtools.js

### Requirements

1. [Redux devtools extension](https://github.com/heldr/redux-devtools-extension) (Really? LOL)
2. Browser must support [Function.name](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name#Browser_compatibility).

### Performance caveats

1. Adds 8 bytes in **unistore.js**
2. Extra function(bind call) when action has promise.

### Example
https://codesandbox.io/s/l49l79r06z

### CODE REVIEW

File: unistore.js [hunk 1]
```diff
@@ -30,8 +30,7 @@ export function createStore(state) {
 
 	function setState(update, overwrite) {
 		state = overwrite ? update : assign(assign({}, state), update);
-		let currentListeners = listeners;
```
Comment (developit) [inline unistore.js]: This line is important, it retains a variable reference locally that is replaced in outer scope.

File: unistore.js [hunk 2]
```diff
@@ -51,14 +50,14 @@ export function createStore(state) {
 			return function() {
 				let args = [state];
 				for (let i=0; i<arguments.length; i++) args.push(arguments[i]);
-				let ret = action.apply(this, args);
+				let ret = action.apply(action, args);
 				if (ret!=null) {
-					if (ret.then) ret.then(setState);
-					else setState(ret);
+					if (ret.then) ret.then(setState.bind(action));
```
Comment (developit) [inline unistore.js]: No need for binding here.

Comment (developit) [reply unistore.js]: Ah I see now - what about passing (state, action) to subscribers? Bind is expensive.

File: unistore.js (hunk 1 above)
Comment (heldr) [reply unistore.js]: That's true!

File: unistore.js [hunk 3]
```diff
@@ -50,10 +51,10 @@ export function createStore(state) {
 			return function() {
 				let args = [state];
 				for (let i=0; i<arguments.length; i++) args.push(arguments[i]);
-				let ret = action.apply(action, args);
+				let ret = action.apply(this, args);
 				if (ret!=null) {
-					if (ret.then) ret.then(setState.bind(action));
-					else setState.call(action, ret);
+					if (ret.then) ret.then(res => setState(res, false, action));
```
Comment (heldr) [inline unistore.js]: results in
```js
(i.then?i.then(function(n){return r(n,!1,t)})
```

**Comment by heldr** [comment]
Just found out another limitation, calling `store.setState()` straight will not reflect in devtools since it doesn't have action argument. Should we present anonymous function in this case?

**Comment by developit** [comment]
@heldr yes - maybe we log a `setState` action?

**Comment by heldr** [comment]
I've added `gzip-size-cli` to devDependencies since it was missing.

File: src/index.js [hunk 4]
```diff
@@ -52,12 +52,12 @@ export default function createStore(state) {
 				for (let i=0; i<arguments.length; i++) args.push(arguments[i]);
 				let ret = action.apply(this, args);
 				if (ret!=null) {
-					if (ret.then) ret.then(setState);
-					else setState(ret);
+					if (ret.then) ret.then(res => setState(res, false, action));
```
Comment (developit) [inline src/index.js]: Idea: we could hoist the action -> setState() proxy to L49:

```js
action(action) {
  function apply(result) {
    setState(result, false, action);
  }
  return function() {
    // ... etc
    if (ret.then) ret.then(apply);
    else apply(ret)
  };
}
```

That should save a few bytes, and it avoids the nested closure + return!

**Comment by heldr** [comment]
Something left to change?

**Comment by developit** [comment]
nope, i think we can do it!

**Pull request merged**